demo/src/app/components/+dropdown/demos/visibility-events/visibility-events.ts

```diff
@@ -0,0 +1,16 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'demo-dropdown-visibility-events',
+  templateUrl: './visibility-events.html'
+})
+export class DemoDropdownVisibilityEventsComponent {
+  messages: string[] = [];
+  message = 'onShown';
+  handler(value: string): void {
+    if(this.message === value) {
+      this.messages = [];
+    }
+    this.messages.push(`Event ${value} is fired`);
```

Comment:
unshift and limit to length of 2-3 events in array
it will simplify your code

Reply:
@valorkin fixed it